tapeagents/finetune/finetune.py

```diff
@@ -155,6 +156,10 @@ def run_finetuning_loop(
         logger.info("LR after loading training state: %.2E" % training_metrics.lr)
         dt = log_time(dt, "finetune/training_state_load")
 
+    if accelerator.is_main_process:
```

Comment:
You have just made it mandatory to make a full pass over an arbitrarily large training set.... not a great default behavior.

Can you implement this as a running metric in the training loop instead? You can report it every time the gradient accumulation is finished.